pip/pip-364.md

```diff
@@ -0,0 +1,476 @@
+
+The larger the load difference between brokers, the smaller the number of times it takes to trigger bundle unloads, 
+which can adapt to scenarios such as cluster rolling restart or expansion.
+
+## placement strategy
+As mentioned earlier, `AvgShedder` bundles the shedding and placement strategies, and a bundle has already determined 
+its next owner broker based on the shedding strategy during shedding. But we not only use placement strategies after
+executing shedding, but also need to use placement strategies to assign bundles during cluster initialization, rolling
+restart, and broker shutdown. So how should we assign these bundles without shedding strategies?
+
+We use a hash allocation method: hash mapping a random number to broker. Hash mapping roughly conforms to 
+a uniform distribution, so bundles will be roughly evenly distributed across all brokers. However, due to the different
+throughput between different bundles, the cluster will exhibit a certain degree of imbalance. However, this problem is 
+not significant, and the subsequent balancing can be achieved through shedding strategies. Moreover, the frequency of 
+cluster initialization, rolling restart, and broker shutdown scenarios is not high, so the impact is slight.
+
+## summary
+In summary, `AvgShedder` can solve the problems of load jitter, heterogeneous environment, slow load balancing, etc.
+Following table summarizes the advantages and disadvantages of the three options:
+
+| Combination                                 | heterogeneous environment | load jitter | over placement problem | over unloading problem | slow load balancing |
+|---------------------------------------------|------------------------|------------|-----------------------|-----------------------|--------------|
+| ThresholdShedder + LeastResourceUsageWithWeight | normal                 | good       | bad                   | bad                   | normal       |
+| UniformLoadShedder + LeastLongTermMessageRate | bad                    | bad        | good                  | good                  | normal       |
+| AvgShedder                                   | normal                 | good       | good                  | good                  | good         |
+
+
+# Detailed Design
+
+### Configuration
+
+To avoid introducing too many configurations when calculating how much traffic needs to be unloaded, `AvgShedder` reuses the
+following three `UniformLoadShedder` configurations:
+```
+    @FieldContext(
+            dynamic = true,
+            category = CATEGORY_LOAD_BALANCER,
+            doc = "In the UniformLoadShedder strategy, the minimum message that triggers unload."
+    )
+    private int minUnloadMessage = 1000;
+
+    @FieldContext(
+            dynamic = true,
+            category = CATEGORY_LOAD_BALANCER,
+            doc = "In the UniformLoadShedder strategy, the minimum throughput that triggers unload."
+    )
+    private int minUnloadMessageThroughput = 1 * 1024 * 1024;
```

Comment:
It would be great if we could make the minUnloadMessageThroughput calculated dynamically. This idea aims to resolve the following point.
When a Pulsar cluster has a huge number of topics/producers/consumers, but with low traffic, it is hard for us to determine the `minUnloadMessageThroughput`. If we change this value too small, and new traffic comes in, the topics will be transferred between brokers with high frequency.

Reply:
Actually, `AvgShedder` can deal with such situation without `minUnloadMessageThroughput`. 
The design of `AvgShedder` comes with high stability.